<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
<script>
// 案例1：
//  网络请求: aaa       ->  自己处理(10行)
//  处理：    aaa111    ->  自己处理(10行)
//  处理：    aaa11122  ->  自己处理(10行)
/*
new Promise((resolve,reject)=>{
  setTimeout(()=>{
    resolve('aaa')
  },1000)
}).then(res=>{
  console.log(res,'第一层的10行代码')

  // 第二次请求
  return new Promise((resolve)=>{
    resolve(res+'111')
  })
}).then(res=>{
  console.log(res,'第二层的10行代码')
 
  // 第二次请求
  return new Promise((resolve)=>{
    resolve(res+'222')
  })
}).then(res=>{
  console.log(res)
})
*/

// 案例1-2： 精简1
// new Promise(resolve => resolve())简写
/*
new Promise((resolve,reject)=>{
  setTimeout(()=>{
    resolve('aaa')
  },1000)
}).then(res=>{
  console.log(res,'第一层的10行代码')

  // 第二次请求
  return Promise.resolve(res+'111')
}).then(res=>{
  console.log(res,'第二层的10行代码')
 
  // 第三次请求
  return Promise.resolve(res+'222')
}).then(res=>{
  console.log(res)
})
*/


//案例1-3： 精简2(内部会进行promise包装)
// new Promise((resolve,reject)=>{
//   setTimeout(()=>{
//     resolve('aaa')
//   },1000)
// }).then(res=>{
//   console.log(res,'第一层的10行代码')

//   // 第二次请求
//   return res+'111'
// }).then(res=>{
//   console.log(res,'第二层的10行代码')
 
//   // 第三次请求
//   return res+'222'
// }).then(res=>{
//   console.log(res)
// })

// 案例2：请求失败的话
// new Promise((resolve,reject)=>{
//   setTimeout(()=>{
//     resolve('aaa')
//   },1000)
// }).then(res=>{
//   console.log(res,'第一层的10行代码')

//   // 第二次请求
//   return new Promise((resolve,reject)=>{
//     reject('error message')
//   })
// }).then(res=>{
//   console.log(res,'第二层的10行代码')
 
//   // 第三次请求
//   return res+'222'
// }).then(res=>{
//   console.log(res)
// }).catch(err=>{
//   console.log(err)
// })

// 案例2-2: 简写
new Promise((resolve,reject)=>{
  setTimeout(()=>{
    resolve('aaa')
  },1000)
}).then(res=>{
  console.log(res,'第一层的10行代码')

  // 第二次请求
  // return Promise.reject('error message')
  throw 'error message'
}).then(res=>{
  console.log(res,'第二层的10行代码')

  // 第三次请求
  return res+'222'
}).then(res=>{
  console.log(res)
}).catch(err=>{
  console.log(err)
})






</script>
</body>
</html>